dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClasspathChecker.java

```diff
@@ -49,33 +49,43 @@
 
   private static final Logger logger = Logger.getLogger(ClasspathChecker.class.getName());
 
-  public static ClasspathChecker create(
-      boolean onlyReachable, List<Path> jarPaths, Iterable<Path> entryPoints)
+  public static ClasspathChecker create(List<Path> jarPaths, Iterable<Path> entryPoints)
       throws IOException {
     Preconditions.checkArgument(
         !jarPaths.isEmpty(),
         "The linkage classpath is empty. Specify input to supply one or more jar files");
     ClassDumper dumper = ClassDumper.create(jarPaths);
-    return new ClasspathChecker(onlyReachable, dumper, entryPoints);
-  }
 
-  /**
-   * If true, the report excludes linkage errors on classes that are not reachable
-   * from the entry points of the class usage graph.
-   */
-  private final boolean reportOnlyReachable;
+    ImmutableMap.Builder<Path, SymbolReferenceSet> jarToSymbolBuilder = ImmutableMap.builder();
+    for (Path jarPath : jarPaths) {
+      jarToSymbolBuilder.put(jarPath, ClassDumper.scanSymbolReferencesInJar(jarPath));
```

Comment:
The logic of scanning jar files and building references has been moved to before constructing `ClasspathChecker`.
This is because the output (class references) is required to create `ClassReferenceGraph` and I wanted to make the `classReferenceGraph` field as `final`, meaning that the class references need to be available before constructing `ClasspathChecker` instance.